### ISSUE TITLE
Better Upload Options

### ISSUE THREAD

**Issue by c0fe** [nextcloud/android#2415]
When uploading files or photos using the Nextcloud Android application the options show only icons and no text whatsoever about this. This is rather confusing since the icons are not obvious to their meaning though this is somewhat difficult to achieve with icons in general.

https://i.imgur.com/hEsLcme.png

I propose adding text to the upload such as here:

https://i.imgur.com/q05qwbL.png

This makes life a lot easier and honestly just looks better, in my honest opinion.

**Comment by stratege1401** [comment]
Agreed. Furthermore, icons don't have the same meaning in different country ... i am not drawing you the idea, but V in US/FRANCE means Victory, but for UK people it means something ... different.

**Comment by tobiasKaminsky** [comment]
You can longpress on the icons and then see a description (which is a commen UX pattern).

![2018-04-03-134112](https://user-images.githubusercontent.com/5836855/38247224-b8239d4e-3744-11e8-976e-9a4e9cbb9621.png)

Also the description is shown until a button is clicked right after first installation of the app, to introduce the meaning to the new user.

![2018-04-03-133723](https://user-images.githubusercontent.com/5836855/38247227-bca2499c-3744-11e8-9325-2e803fa1e1a3.png)

This should be enough, or?

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky I'd actually agree with the proposed solution. I checked dropbox and also GDrive and both storage cloud apps do it like the proposal by showing a modal bottom sheet (see also https://material.io/guidelines/components/bottom-sheets.html#bottom-sheets-modal-bottom-sheets) 
![device-2018-04-03-134720](https://user-images.githubusercontent.com/1315170/38247551-07e810f2-3746-11e8-908b-2857d414bc86.png)
![device-2018-04-03-134738](https://user-images.githubusercontent.com/1315170/38247556-0a64053e-3746-11e8-8ff2-8789af9f6771.png)

What do you think @nextcloud/designers ?

**Comment by c0fe** [comment]
My vote is for Material design one, if my vote matters at all :D

**Comment by AndyScherzinger** [comment]
@c0fe which one do you mean since all of them including the actual solution present in the app are material design 😁

**Comment by c0fe** [comment]
@AndyScherzinger I meant the Google Drive one, the design where they are separated out appears to be better.

**Comment by jancborchardt** [comment]
Definitely good point, the text should always be visible.

As a quick fix we should show the text always, not only on first start.

As a second step, we could move to a bottom sheet – like the one from Dropbox.

**Comment by c0fe** [comment]
Perhaps it would be better to have the options

- Upload Images
- Upload Files
- Create Folder

I do not think "Content from Applications" is necessary since you could upload it directly from there though overall the option just sounds confusing.

**Comment by AndyScherzinger** [comment]
Working on it :)

**Comment by c0fe** [comment]
@AndyScherzinger is there a beta channel for the nextcloud app? like imgur has a beta channel for its app?

**Comment by AndyScherzinger** [comment]
@c0fe yes there are several...

* On f-droid release candidates are published but not propagated actively by f-droid so you'd need to manually check the app for updates within the f-droid app
* On f-droid there are nightly builds of the app: https://f-droid.org/en/packages/com.nextcloud.android.beta/
On GPlay there is the beta channel, opt-in via https://play.google.com/apps/testing/com.nextcloud.client

**Comment by AndyScherzinger** [comment]
PR is #2432

**Comment by c0fe** [comment]
@AndyScherzinger thanks. I signed up just now and will annoy you guys with more reports of bugs.

What about scan document? Sort of like what CamScanner has, this would be a useful feature, would this be a separate issue request?

**Comment by AndyScherzinger** [comment]
This would be something we likely wouldn't implement (just my personal opinion!), since it is already covered by other apps that support the DAV protocol, see https://medium.com/@mathiasconradt/using-scanbot-and-nextcloud-for-scanning-ocr-and-file-management-229750420882 for example.

But in general it would be another issue to  be opened and discussed :-)

**Comment by jancborchardt** [comment]
Very nice work by the way @AndyScherzinger, much better in day-to-day use. :)

**Comment by AndyScherzinger** [comment]
Thanks @jancborchardt :heart: Always happy to improve the App from a behavior/UX/interaction/you-name-it point of view :)

**Comment by c0fe** [comment]
@AndyScherzinger is there a beta version of this and if so how can I sign for it or do I have to just compile on PC and upload the APK?

**Comment by AndyScherzinger** [comment]
@c0fe yes there is.

* For Google Play users: You need to register here https://play.google.com/apps/testing/com.nextcloud.client as a tester and than GPlay will show you the beta version 
* For f-droid users: Search for the app in the f-droid client since f-droid does offer the Release Candidates but doesn't actively notify you about them as-in App Update Notifications, see also https://f-droid.org/packages/com.nextcloud.client/

Besides the Betas as an f-droid user you could also checkout the dev builds (can by installed in parallel to the stable releases!): https://f-droid.org/packages/com.nextcloud.android.beta/
